### PR TITLE
[clang-tidy] Remove enforcement of rule C.48 from cppcoreguidelines-p…

### LINKED ISSUE #62169: [clang-tidy] Remove enforcement of rule C.48 from cppcoreguidelines-prefer-member-init
This functionality has been deprecated, moved to cppcoreguidelines-use-default-member-init, which is an alias of modernize-use-default-member-init. Remove the existing functionality in clang-tidy 19.

### PR DESCRIPTION
**Pull request by carlosgalvezp**
…refer-member-init

This functionality already exists in
cppcoreguidelines-use-default-member-init. It was deprecated from this check in clang-tidy 17.

This allows us to fully decouple this check from the corresponding modernize check, which has an unhealthy dependency.

Fixes #62169

### CODE REVIEW

**Review by EugeneZelenko — APPROVED** [review]

File: clang-tools-extra/clang-tidy/cppcoreguidelines/PreferMemberInitializerCheck.cpp [hunk 1]
```diff
@@ -230,139 +197,99 @@ void PreferMemberInitializerCheck::check(
...
-          {UseAssignment ? " = " : "{",
-           Lexer::getSourceText(Result.SourceManager->getExpansionRange(
-                                    InitValue->getSourceRange()),
-                                *Result.SourceManager, getLangOpts()),
-           UseAssignment ? "" : "}"});
-
-      Diag << FixItHint::CreateInsertion(FieldEnd, Insertion)
-           << FixItHint::CreateRemoval(StmtRange);
-
-    } else {
```
Comment (carlosgalvezp) [inline clang-tools-extra/clang-tidy/cppcoreguidelines/PreferMemberInitializerCheck.cpp:272]: The diff is very unfortunate, but essentially there's no changes to the code inside this "else" block. It's just unindented 2 spaces. The main change is the removal of the "if" block.

**Review by PiotrZSL — APPROVED** [review]
Looks fine.

For me check names are not the best, it's hard to figure out what they does. Maybe they should be named like prefer-using-initialization-list and prefer-using-in-class-initializers.
Simply because now member initializer could be both an in class and on initialization list.

Second thing is that would be nice if changed check name is more early in release notes, like "Changed/Improved cppcoreguidelines-prefer-member-initializer by removing enforcement of rule C.48, , deprecated since :program:clang-tidy 17. This rule is now covered by cppcoreguidelines-use-default-member-init check."

I didn't check too deeply changed, code after all that's mainly code removal, and that's covered by tests.

**Comment by carlosgalvezp** [comment]
Thanks for the review! I agree, check names are not the best. It's a bit of a hassle to change them though :/ 

I will fix the comment about the release notes tonight :+1:

**Comment by carlosgalvezp** [comment]
I've screwed up with the merge conflicts somehow, will create a new PR

**Pull request closed without merging**